driver/src/main/java/oracle/nosql/driver/ops/PutRequest.java

```diff
@@ -437,6 +438,31 @@ public int getIdentityCacheSize() {
         return identityCacheSize;
     }
 
+    /**
+     * Returns the metadata of the row to be used.
+     *
+     * @return the metadata, or null if not set
+     *
+     * @since 5.4.18
+     */
+    public String getRowMetadata() {
+        return rowMetadata;
+    }
+
+    /**
+     * Sets the row metadata to use for the put operation.
```

Comment:
Needs to discuss constraints on the string, possible errors thrown, and add JSON validation (cheap) code

Reply:
Done.